lib/foreman_maintain/upgrade_runner.rb

```diff
@@ -34,32 +34,25 @@ def clear_current_target_version
 
     attr_reader :version, :tag, :phase
 
-    def initialize(version, reporter, options = {})
+    def initialize(reporter, options = {})
       super(reporter, [], options)
-      @version = version
-      scenarios_present = find_scenarios(:tags => :upgrade_scenario).any?(&matching_version_test)
-      raise "Unknown version #{version}" unless scenarios_present
-
       @scenario_cache = {}
-      self.phase = :pre_upgrade_checks
+      @phase = :pre_upgrade_checks
+      condition = { :tags => [:upgrade_scenario, phase] }
+      matching_scenarios = find_scenarios(condition)
+      @version = matching_scenarios.first.target_version
```

Comment:
if next scenario is not installed (next maintenance is not enabled) and you run "upgrade" then it produces:
```
undefined method `target_version' for nil:NilClass
```
Is question mark missing?
```suggestion
      @version = matching_scenarios.first.target_version?
```

Reply:
```suggestion
      @version = matching_scenarios.first&.target_version
```

Safe operator is `&`, but I wonder what else that might break

Reply:
```
[root@sat-stream-qa-rhel8 ~]# foreman-maintain upgrade check
Checking for new version of satellite-maintain...
Nothing to update, can't find new version of satellite-maintain.
[root@sat-stream-qa-rhel8 ~]# foreman-maintain upgrade run
Checking for new version of satellite-maintain...
Nothing to update, can't find new version of satellite-maintain.
--------------------------------------------------------------------------------
Upgrade finished.
```

This doesn't look correct to me. I'd have expected an error "there is nothing to upgrade to" or something like that.